### ISSUE TITLE
Error when mocking core modules with the `node:` prefix (minor annoyance)

### ISSUE THREAD

**Issue by gmahomarf** [iambumblehead/esmock#115]
Context:
```.js
// someModule.js

import path from 'node:path';

...
```

Trying to do something like the following:
```.js
const someModule = await esmock('./someModule.js', {
    'node:path': pathMock,
});
```

Results in the following error:
```
Error: not a valid path: "node:path"
```

This works without changing `someModule.js`: 
```.js
const someModule = await esmock('./someModule.js', {
    'path': pathMock,
});
```

However, now the module URLs in the tests and the original module are different. I've opened a PR in iambumblehead/resolvewithplus#27 to add support for the `node:` prefix.

**Comment by iambumblehead** [comment]
@gmahomarf thanks for the great PR which will be published soonish https://github.com/iambumblehead/esmock/pull/116

This temporary work-around was needed for now. The long-term solution is for resolvewithplus to return the final path always with the protocol like import.meta.resolve would. The long-term solution can probably be added before next-week,
```diff
diff --git a/src/esmockModule.js b/src/esmockModule.js
index 1df6117..297a9ae 100644
--- a/src/esmockModule.js
+++ b/src/esmockModule.js
@@ -18,7 +18,9 @@ const FILE_PROTOCOL = 'file:///'
 // https://url.spec.whatwg.org/, eg, file:///C:/demo file:///root/linux/path
 const pathAddProtocol = (pathFull, protocol) => {
   if (!protocol)
-    protocol = !resolvewith.iscoremodule(pathFull) ? FILE_PROTOCOL : 'node:'
+    protocol = /^node:/.test(pathFull)
+      ? ''
+      : !resolvewith.iscoremodule(pathFull) ? FILE_PROTOCOL : 'node:'
   if (protocol.includes(FILE_PROTOCOL))
     pathFull = fs.realpathSync.native(pathFull)
   if (process.platform === 'win32')
```

**Comment by iambumblehead** [comment]
it is published to [esmock@1.9.4](https://www.npmjs.com/package/esmock)

**Comment by iambumblehead** [comment]
this seems to be resolved but feel free to reopen if necessary
